bindata/network/openshift-sdn/alert-rules.yaml

```diff
@@ -12,22 +12,25 @@ spec:
   groups:
   - name: general.rules
     rules:
+    # note: all joins on kube_pod_* need a a "topk by (key) (1, <metric> )"
+    # otherwise you will generate query errors when kube_state_metrics is being
+    # upgraded and there are duplicate rows on the "right" side of the join.
     - alert: NodeWithoutOVSPod
       annotations:
         message: |
           All nodes should be running an ovs pod, {{"{{"}} $labels.node {{"}}"}} is not.
       expr: |
-        (kube_node_info unless on(node) kube_pod_info{namespace="openshift-sdn",  pod=~"ovs.*"}) > 0
-      for: 20m
+        (kube_node_info unless on(node) topk by (node) (1, kube_pod_info{namespace="openshift-sdn",  pod=~"ovs.*"})) > 0
+      for: 10m
       labels:
         severity: warning
     - alert: NodeWithoutSDNPod
       annotations:
         message: |
           All nodes should be running an sdn pod, {{"{{"}} $labels.node {{"}}"}} is not.
       expr: |
-        (kube_node_info unless on(node) kube_pod_info{namespace="openshift-sdn",  pod=~"sdn.*"}) > 0
-      for: 20m
+        (kube_node_info unless on(node) topk by (node) (1, kube_pod_info{namespace="openshift-sdn",  pod=~"sdn.*"})) > 0
+      for: 10m
       labels:
         severity: warning
     - alert: NetworkPodsCrashLooping
```

Comment:
We have an alert for pods crashlooping, so not sure this one is needed? https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/37f8263d17bc9d429fece3d345365650587bc56c/alerts/apps_alerts.libsonnet#L24

Reply:
oh, good point, removed.